tests/core_tests.py

```diff
@@ -404,6 +405,219 @@ def test_only_owners_can_save(self):
+        self.assertEquals(
+            '[main].[wb_health_population](id:{})'.format(table_id),
+            slc_2.perm)
+
+        # Case 3. Import slice with no permissions
+        slice_3 = get_slice('Import Me 3', slc_id=3)
+        slice_3.perm = None
+        slc_id_3 = models.Slice.import_slice(slice_3)
+        self.assertIsNone(slc_id_3)
+
+        # Case 4. Import slice with non existent datasource
+        slc_id_4 = models.Slice.import_slice(get_slice(
+            'Import Me 4', slc_id=4, perm='[main].[non_existent](id:999)'))
+        self.assertIsNone(slc_id_4)
+
+        # Case 5. Slice already exists, no override
+        slc_id_5 = models.Slice.import_slice(
+            get_slice('Import Me 1'), import_time=1989)
+        slc_5 = find_slice(slc_id_5)
+        self.assertEquals(table_id, slc_5.datasource_id)
+        self.assertEquals(
+            '[main].[wb_health_population](id:{})'.format(table_id),
+            slc_5.perm)
+        self.assertEquals('Import Me 1', slc_5.slice_name)
+        self.assertEquals('table', slc_5.datasource_type)
+        self.assertEquals('bubble', slc_5.viz_type)
+        self.assertEquals(
+            {"num_period_compare": "10", "remote_id": 1, "import_time": 1989},
+            json.loads(slc_5.params))
+
+        # Case 6. Older version of slice already exists, override
+        slc_id_6 = models.Slice.import_slice(
+            get_slice('Import Me New 1'), import_time=1990)
+        slc_6 = find_slice(slc_id_6)
+        self.assertEquals(table_id, slc_6.datasource_id)
+        self.assertEquals(
+            '[main].[wb_health_population](id:{})'.format(table_id),
+            slc_5.perm)
+        self.assertEquals('Import Me New 1', slc_6.slice_name)
+        self.assertEquals('table', slc_6.datasource_type)
+        self.assertEquals('bubble', slc_6.viz_type)
+        self.assertEquals(
+            {"num_period_compare": "10", "remote_id": 1, "import_time": 1990},
+            json.loads(slc_6.params))
+
+        session.delete(slc_2)
+        session.delete(slc_6)
+        session.commit()
+
+    def test_import_dashboard(self):
+        session = db.session
+
+        def get_dashboard(title, id=0, slcs=[]):
+            return models.Dashboard(
+                id=id,
+                dashboard_title=title,
+                slices=slcs,
+                position_json='{"size_y": 2, "size_x": 2}',
+                slug='{}_imported'.format(title.lower()),
+            )
+
+        def find_dash(dash_id):
+            return session.query(models.Dashboard).filter_by(
+                id=dash_id).first()
+
+        def get_slc(name, table, slc_id):
+            return models.Slice(
+                id=slc_id,
+                slice_name=name,
+                datasource_type='table',
+                perm='[main].[{}](id:{})'.format(table, slc_id)
+            )
+
+        # Case 1. Import empty dashboard
+        empty_dash = get_dashboard('empty_dashboard', id=1001)
```

Comment:
I'd rather have it as multiple smaller tests, ideally not in `core_tests.py` but in a new `import_export.py`